packages/frontend/src/providers/AppProvider.test.tsx

```diff
@@ -0,0 +1,34 @@
+import { renderHook, waitFor } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import { renderHookWithProviders } from '../testing/testUtils';
+import { useApp } from './AppProvider';
+
+describe('AppProvider', () => {
+    it('should throw if used outside of AppProvider', () => {
+        expect(() => {
+            renderHook(() => useApp());
+        }).toThrow('useApp must be used within a AppProvider');
+    });
+
+    it('should return values from AppProvider', () => {
+        expect(async () => {
+            const { result } = renderHookWithProviders(() => useApp());
```

Comment:
`renderHookWithProviders` is very similar to `renderWithProviders`. it also accepts overrides for mocks (user, health) as the second argument!